### PR TITLE
Code component

### PR DESCRIPTION
**Pull request by khanzadyan**
Explanation of code component. The screenshot needs an update.

### CODE REVIEW

**Comment by khanzadyan** [comment]
@jhorbulyk ready for a review

File: _components/code-component.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+*   `exports`: {},
+*   `messages`: - [Utility for convenient message creation](https://github.com/elasticio/elasticio-node/blob/master/lib/messages.js),
+*   `request`: - [Http Client](https://github.com/id0x3d/co-request),
+*   `wait`: - wait,
+*   `emitter`: user to emit messages and errors
+
+
+## Known limitations
+
+Only the code written in `Node.js` is supported at this moment.
```
Comment (jhorbulyk) [inline _components/code-component.md]: No `async` support.

File: _components/code-component.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
+---
+title: Code component
+layout: article
+section: Utility Components
+---
+
+## Description
+
+{{site.data.tenant.name}} Code component executes a piece of code. It is available
```
Comment (jhorbulyk) [inline _components/code-component.md]: Have you seen some update work done by @zubairov https://github.com/elasticio/code-component/pull/9 ?

File: _components/code-component.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+*   `process`: - [Current Node.js process](https://nodejs.org/dist/latest-v5.x/docs/api/process.html),
+*   `require`: - [Module require](https://nodejs.org/dist/latest-v5.x/docs/api/modules.html),
+*   `setTimeout`: - more on [setTimeout](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+*   `clearTimeout`: - more on [clearTimeout](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+*   `setInterval`: - more on [setInterval](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+*   `clearInterval`: - more on [clearInterval](https://nodejs.org/dist/latest-v5.x/docs/api/timers.html),
+*   `msg`: - Incoming message containing the payload from the previous step,
+*   `exports`: {},
+*   `messages`: - [Utility for convenient message creation](https://github.com/elasticio/elasticio-node/blob/master/lib/messages.js),
+*   `request`: - [Http Client](https://github.com/id0x3d/co-request),
```
Comment (jhorbulyk) [inline _components/code-component.md]: Add a remark that this is `co-request`.

File: _components/code-component.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+*   `wait`: - wait,
+*   `emitter`: user to emit messages and errors
+
+
+## Known limitations
+
+Only the code written in `Node.js` is supported at this moment.
+
+## Code component usage Examples
```
Comment (jhorbulyk) [inline _components/code-component.md]: Most of these methods are super-dated, but then so is the component.  Perhaps limit it to the callback section and scrap the other two sections?

File: _components/code-component.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    });
+
+    // Emit the data event
+    emitter.emit('data', data);
+
+    console.log('Message emitted');
+});
+```
+
+For a simple use case, this is sufficient, however, if we then decide to include
```
Comment (jhorbulyk) [inline _components/code-component.md]: My suggestion is to remove this paragraph.  I don't think it is useful to transition from explaination of component to how to code JS tutorial here.

File: _components/code-component.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+        email: body.email,
+        userID: body.id
+    });
+
+    // Emit the data event
+    emitter.emit('data', data);
+
+    console.log('Message emitted');
+});
+```
```
Comment (jhorbulyk) [inline _components/code-component.md]: Can you double check that this works?

File: _components/code-component.md (hunk 1 above)
Comment (jhorbulyk) [reply _components/code-component.md]: Can you add some remarks that currently this component does not support snapshots or any form of credentials?

**Comment by khanzadyan** [comment]
@jhorbulyk I will redo this after the README by @zubairov .

**Comment by khanzadyan** [comment]
Closing the pull, it is obsolete

**Pull request closed without merging**